### PR TITLE
[build] Run Venice CI with GitHub Actions

### PR DESCRIPTION
**Pull request by sushantmane**
Run Venice tests on the following matrix:
OS: macOS, Ubuntu
JDK: 11

## Run Venice CI with GitHub Actions

While we figure out how we want to run CI with GitHub actions, I think it would be good to have
something up and running in the meantime. With this RB, CI will be run on every commit/pull request. 
It triggers one job/OS for unit tests (~42 mins) and one for integration tests/OS (2.30 hours and a bit flaky). 

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Comment by FelixGV** [comment]
Thanks Sushant! Is it normal that the checks show as failed?

**Comment by sushantmane** [comment]
> Thanks Sushant! Is it normal that the checks show as failed?

No, checks should not fail. I know the reason: this PR ('get fetch' part in actions) was designed to work on forks and I need to update it to work everywhere. I'll update it. Thanks.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/utils/IsolatedIngestionUtils.java [hunk 1]
```diff
@@ -308,7 +308,11 @@ public static void releaseTargetPortBinding(int port) {
    */
   public static Optional<Integer> getLingeringIngestionProcessId(int port) {
     Optional<Integer> isolatedIngestionProcessPid = Optional.empty();
-    String processIds = executeShellCommand("/usr/sbin/lsof -t -i :" + port);
```
Comment (nisargthakkar) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/utils/IsolatedIngestionUtils.java:311]: Why do we need the `/usr/sbin` prefix? It should be in the path by default

File: .github/workflows/gh-ci.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            git remote add upstream https://github.com/linkedin/venice
+            git fetch upstream
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+        with:
+          arguments: "clean check -x spotlessCheck --continue --no-daemon -Pspotallbugs -Pspotbugs.xml -Pspotbugs.ignoreFailures -x integrationTest -x test"
+
+  UnitTests:
```
Comment (nisargthakkar) [inline .github/workflows/gh-ci.yml:30]: Can you also add a spotbugs step?

Comment (nisargthakkar) [reply .github/workflows/gh-ci.yml:30]: I see it's already there. Removing

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/utils/IsolatedIngestionUtils.java (hunk 1 above)
Comment (sushantmane) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/utils/IsolatedIngestionUtils.java:311]: A full path is required for CentOS/RHEL. The path is not the same on Ubuntu and it works without the path on it.  

Also, GH actions runner doesn't have CentOS and hence I'm using Ubuntu.

**Comment by FelixGV** [comment]
Hi Sushant, would you mind rebasing and retrying this? Curious to see how it goes now. Thanks!

**Comment by FelixGV** [comment]
I think we need to divide the e2e tests further. One first cut might be to do a split along the lines of:

1. Run all integration tests that match `com.linkedin.venice.endToEnd.*`.
2. Run all integration tests that don't match `com.linkedin.venice.endToEnd.*`

Might need to define a couple more test targets in the Gradle build. This seems useful for that: https://mkyong.com/gradle/gradle-how-to-exclude-some-tests/

Are you game to give it a try?

**Comment by sushantmane** [comment]
> I think we need to divide the e2e tests further. One first cut might be to do a split along the lines of:
> 
> 1. Run all integration tests that match `com.linkedin.venice.endToEnd.*`.
> 2. Run all integration tests that don't match `com.linkedin.venice.endToEnd.*`
> 
> Might need to define a couple more test targets in the Gradle build. This seems useful for that: https://mkyong.com/gradle/gradle-how-to-exclude-some-tests/
> 
> Are you game to give it a try?

> 

Sorry for the late response. Sure, I'll split e2e tests into two parts. They used to finish in [< 3HR ](https://github.com/sushantmane/venice/actions/runs/3521681573/jobs/5903795730) and hence I added 3HR timeout. But it seems we've added >25 e2e tests since then.

**Comment by sushantmane** [comment]
@FelixGV - 
> 1. Run all integration tests that match com.linkedin.venice.endToEnd.*.

Completes in 2h10m

> 2. Run all integration tests that don't match com.linkedin.venice.endToEnd.*

Completes in 1h

@adamxchen  is planning to work on this in the next quarter. In the meantime, should we use this PR?

**Comment by FelixGV** [comment]
Thanks a lot for the initiative Sushant!! I think we should merge this in, for sure!

In the future, I would like us to reduce the wall clock time by splitting further, and/or deleting redundant tests.

In any case, this is a great start! Thanks again 🙏

**Review by FelixGV — APPROVED** [review]
Tremendous 🤩🍾🙏

**Comment by sushantmane** [comment]
Thanks for the review. I'll run our internal CI once just to make sure everything else is alright.

**Comment by sushantmane** [comment]
@FelixGV - Could you please approve the PR again?

**Review by nisargthakkar — APPROVED** [review]
LGTM

**Review by FelixGV — APPROVED** [review]
Thanks a lot! Please make sure to review and clean up the commit message while squashing.

**Comment by sushantmane** [comment]
Thank you @FelixGV & @nisargthakkar!

**Pull request merged**